scipy/cluster/tests/test_hierarchy.py

```diff
@@ -200,6 +200,25 @@ def test_linkage_cophenet_tdist_Z_Y(self, xp):
         xp_assert_close(c, expectedc, atol=1e-10)
         xp_assert_close(M, expectedM, atol=1e-10)
 
+    def test_gh_22183(self, xp):
```

Comment:
Also currently this only test numpy as it's an array like input. We should either skip the other backends or use them

Reply:
@j-bowhay thanks, I think I addressed both of your comments in the latest commit.

`python dev.py test -t scipy/cluster/tests/test_hierarchy.py::TestCopheneticDistance::test_gh_22183 -b all` seems "ok" locally on ARM Mac with `numpy`, `torch`, and `array_api_strict` backends.